src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -377,6 +377,39 @@ def clone_status(self, **kwargs):
             ret = self.volume_exception_to_retval(ve)
         return ret
 
+    def clone_cancel(self, **kwargs):
+        ret       = 0, "", ""
+        volname   = kwargs['vol_name']
+        clonename = kwargs['clone_name']
+        groupname = kwargs['group_name']
+
+        try:
+            with open_volume(self, volname) as fs_handle:
+                with open_group(fs_handle, self.volspec, groupname) as group:
+                    with open_subvol(fs_handle, self.volspec, group, clonename,
+                                     need_complete=False, expected_types=["clone"]) as subvolume:
+                        status = subvolume.status
+                        if not subvolume.is_clone_cancelable(status['state']):
+                            raise VolumeException(-errno.EINVAL, "cannot cancel -- clone finished (check clone status)")
+                        with open_group(fs_handle, self.volspec, status['source'].get('group', None)) as s_group:
+                            with open_subvol(fs_handle, self.volspec, s_group, status['source']['subvolume']) as s_subvolume:
+                                jid = s_subvolume.try_clone_cancel(subvolume, status['source']['snapshot'])
+                                if not jid:
+                                    return ret
+            # clone is in progress -- open the volume in lockless mode and initiate
+            # cancel. volume needs to be open()'d in lockless mode since async cloner
```

Comment:
Cancelling an on-going clone would persist "canceled" state in subvolume metadata. To persist the new state, async cloner accesses the volume in exclusive mode. Accessing the volume in exclusive mode here would lead to deadlock.

Reply:
Thanks. Can you include this explanation in the comment?

Reply:
sure, when I update the pr.